### ISSUE TITLE
type hints are missing in constructors when autodoc_typehints = 'description'

### ISSUE THREAD

**Issue by xuhdev** [sphinx-doc/sphinx#8689]
**Describe the bug**
Type hints are missing in constructors when `autodoc_typehints = 'description'`.

**To Reproduce**

1. Download [bug.tar.gz](https://github.com/sphinx-doc/sphinx/files/5823430/bug.tar.gz) and unarchive.
2. Run `PYTHONPATH=. sphinx-build -d "./_build" docs "_build/html" -b html -W --color`

```
├── docs
│   ├── conf.py
│   └── index.rst
└── myproject
    └── __init__.py
```

conf.py:

```python
project = 'myproject'
copyright = '2020, someone'
author = 'someone'
autodoc_typehints = 'description'
extensions = [
    'sphinx.ext.autodoc',
]
html_theme = 'alabaster'
```

`myproject/__init__.py`:

```python
class A:
    """
    :param boolean: I'm a boolean.
    """
    def __init__(self, boolean: bool):
        pass
```

`index.rst` is minimal:

```rst
Welcome to myproject's documentation!
=====================================

.. automodule:: myproject
   :members:
   :imported-members:
```

**Expected behavior**
Constructor parameter type hints should be documented.

**Screenshots**

With `autodoc_typehints = 'description'`:

![Screenshot_2021-01-15_15-54-22](https://user-images.githubusercontent.com/325476/104789479-fcc61080-5749-11eb-9b90-6228dc7ad501.png)

The type hint of `bool` should be in the description but it's missing.

Without `autodoc_typehints = 'description'`, `bool` is shown on the function prototype line as expected:

![Screenshot_2021-01-15_15-55-14](https://user-images.githubusercontent.com/325476/104789512-1bc4a280-574a-11eb-8d81-31c24cc153cd.png)

**Environment info**
- OS: Debian 10
- Python version: 3.7.6
- Sphinx version: 3.4.3
- Sphinx extensions:  autodoc

**Comment by tk0miya** [comment]
Thank you for reporting. This will be fixed by https://github.com/sphinx-doc/sphinx/pull/8539.

**Comment by tk0miya** [comment]
Now #8539 is merged. It will be released as 4.0 soon.
Thanks,
